pkg/library/flatten/helper.go

```diff
@@ -64,3 +65,14 @@ func formatImportCycle(end *resolutionContextTree) string {
 	}
 	return cycle
 }
+
+func getOriginalNameForComponent(component dw.Component) string {
```

Comment:
Well, we don't modify components names, it returns exactly what is written in body: SourceComponent. Maybe then rename it to `getSourceForComponent`

Reply:
We don't modify component names per se, but we delete plugin components and add e.g. container components:
```yaml
components:
  - name: my-plugin
    plugin: [...]
```
becomes
```yaml
components:
  - name: plugin-container
    container: [...]
```
But I see your point -- I'll update.